index.js

```diff
@@ -123,7 +123,9 @@ function parserForArrayFormat(options) {
 		case 'separator':
 			return (key, value, accumulator) => {
 				const isArray = typeof value === 'string' && value.split('').indexOf(options.arrayFormatSeparator) > -1;
-				const newValue = isArray ? value.split(options.arrayFormatSeparator).map(item => decode(item, options)) : value === null ? value : decode(value, options);
+				const isEncodedArray = (typeof value === 'string' && !isArray && decode(value, options).split('').indexOf(options.arrayFormatSeparator) > -1);
```

Comment:
I don't see the point in splitting the string into an array. You can just look for the character in the string. Also use `Array#includes` instead.